fix(finance): update iso-3166-1 country codes for IBAN/BIC

bmenant opened the pull request:
I had posted the issue `#1261` on the original repository, followed by a pull request ([from this branch](https://github.com/bmenant/faker.js/tree/fix/issue-1261-iban-bic-eu-country-code)).

To put it short: this simply update the list of ISO-3166-1 country codes [1] used for IBAN & BIC generations. As a matter of fact, some codes were nonexistent in the official specifications. Those have caused Faker.js to generate invalid IBAN and BIC codes (notably when checked against [validator.js](https://github.com/validatorjs/validator.js), for example).

~~FYI, as of today, I’m unable to run tests: `Error: Cannot find module './lib'`~~

[1] Both BIC ([ISO-9362](https://en.wikipedia.org/wiki/ISO_9362)) and IBAN ([ISO-13616](https://en.wikipedia.org/wiki/International_Bank_Account_Number)) explicitly refers to ISO-3166-1.

-----

It has also been suggested (here: #131) to use [fake-numbers](https://github.com/phuoc-ng/fake-numbers) for IBAN generation. AFAIK, it uses a [restricted list of country codes](https://github.com/phuoc-ng/fake-numbers/blob/master/src/iban/fake.ts#L21-L43) for IBAN generation (which I believe to be just fine). However, for BIC generation, fake-numbers uses a [random string of 2 letters](https://github.com/phuoc-ng/fake-numbers/blob/master/src/bic/fake.ts#L15) as a country code (which are unlikely to ever produce a valid BIC code IMHO).

Shinigami92 commented:
To tun test you have to run `npm run build` to generate lib content

bmenant commented:
Rebased on [v6.0.0-alpha.5](https://github.com/faker-js/faker/tree/v6.0.0-alpha.5) and updated tests.

bmenant commented on src/iban.ts:
```diff
@@ -1166,7 +1164,6 @@ export = {
     'BD',
     'BE',
     'BF',
-    'BG',
```
True! Looks like the sources I’ve used were corrupted. My bad, I should have double checked. 
Good call @ST-DDT, thank you!

I’ve pushed a fix. I can’t connect iso.org at the moment (maintenance on going), but Wikipedia says there’s currently 249 countries in the list. However, https://github.com/michaelwittig/node-i18n-iso-countries and https://github.com/annexare/Countries counts 250 countries. As a matter of fact, they included `XK` (User-Assigned code used by some organizations for Kosovo). It is a valid country code for IBAN.

🙇 My sincere apologizes to Bulgaria 🇧🇬 and Kosovo 🇽🇰

bmenant commented:
I think it’s out of scope over here, but you might consider refactoring countries data (I suppose there are other places Faker uses these). 

You might also consider adding a dependencies for those, such as [unicode-org/cldr-json](https://github.com/unicode-org/cldr-json), [annexare/Countries](https://github.com/annexare/Countries) or [node-i18n-iso-countries](https://github.com/michaelwittig/node-i18n-iso-countries).

ST-DDT commented:
The ISO page seems to still be down. I will review it once it is back up again.

hertzg commented:
@ST-DDT FYI ISO page seems to be back up

ST-DDT reviewed (approved):
XK doesn't appear in the iso-list, but i found multiple references to it in some other online references, which I would consider good enough.
I also checked the code regarding the usage of the iso codes and it is only used to generate `bic`s so it should be safe.

Shinigami92 commented on test/finance.spec.ts:
```diff
@@ -386,6 +386,9 @@ describe('finance', () => {
       expect(() => faker.finance.iban(false, 'AA')).toThrowError(
```
... not able to address line 385, but could you use the `it.each` functionality to test multiple non supported country codes?

bmenant commented:
Rebased and updated tests as suggested by @Shinigami92 

I had got a new failure in finance.spec.ts in a seed/bic test. I don’t know where those seeded values come from since they are hard-coded, but I had to change one of the BIC code (`LT` to `LY`). Since the order and the number of country codes has changed, it is likely that the country code selected for this particular seeding value has changed as well. Please let me know if I’m wrong.

Shinigami92 commented:
> (`LT` to `LY`). 

This happened because you deleted some entries from the definitions, so the indexes changed. The values are "fetched" from the definitions via `arrayElement` which uses `datatype.number` as index.
So everything is okay 👌

bmenant commented:
@Shinigami92 You reassigned myself, is there anything I can do so far? :slightly_smiling_face:

Shinigami92 commented:
Nope, we just may have another workflow then you expect
We have a big issue/pr board: https://github.com/orgs/faker-js/projects/2/views/9
And there we can see who is or has worked on something, it's mostly the same as the author of a PR
